src/Multi_filtration/include/gudhi/Multi_critical_filtration.h

```diff
@@ -37,400 +40,573 @@ namespace Gudhi::multi_filtration {
 
  */
 template <typename T, bool co = false>
-class Multi_critical_filtration {
+class Multi_critical_filtration
+{
  public:
-  using value_type = T;
-  using OneCritical = One_critical_filtration<T>;  // Type of the One critical subtype
-  template <typename value_type>
-  using Base = Multi_critical_filtration<value_type>;
-  std::size_t num_parameters() const { return multi_filtration_.size() == 0 ? 0 : multi_filtration_[0].size(); }
-  Multi_critical_filtration() : multi_filtration_(1) {
-    multi_filtration_[0] = co ? One_critical_filtration<T>::inf() : One_critical_filtration<T>::minus_inf();
-  };  // initialize at 1, with 0 parameter (consistent with below)
-  Multi_critical_filtration(int n)
-      : multi_filtration_({One_critical_filtration<T>(n)}) {};  // minus infinity by default
-  Multi_critical_filtration(int n, T value)
-      : multi_filtration_({One_critical_filtration<T>(n, value)}) {};  // std::vector<T>(n, value){};
-  Multi_critical_filtration(std::initializer_list<T> init)
-      : multi_filtration_({One_critical_filtration<T>(init)}) {};  // : std::vector<T>({std::vector<T>(init)};
+  using Single_point = One_critical_filtration<T>;  // Type of the One critical subtype
```

Comment:
Me neither, but I needed something else than `Base` and it should also work for `One_critical_filtration`. `Generator` sounds fine if you don't think it could be misunderstood with some other notion of generator in this context.